propagator/opentelemetry-propagator-b3/src/opentelemetry/propagators/b3/__init__.py

```diff
@@ -97,6 +97,8 @@ def extract(
             or self._trace_id_regex.fullmatch(trace_id) is None
             or self._span_id_regex.fullmatch(span_id) is None
         ):
+            if context is None:
+                return trace.set_span_in_context(trace.INVALID_SPAN, context)
             return context
```

Comment:
This should also fix the composite propagator issue. While you are at it, could you also fix another issue with B3 when there is a valid propagation header. `context` should be passed in this `set_span_in_context` call.
https://github.com/open-telemetry/opentelemetry-python/blob/fba4e613fc4a060f3fce9721d88effa50d6f2c31/propagator/opentelemetry-propagator-b3/src/opentelemetry/propagators/b3/__init__.py#L114-L125

Reply:
Done. Also added tests to ensure the returned context is always derived from the provided one.